Reference/Searching/Examine/Indexing/index-v8.md

```diff
@@ -0,0 +1,352 @@
+    public override IValueSetValidator GetMemberValueSetValidator()
+    {
+        // all built in member properties to exclude (i.e. Password question, etc...)
+        var excludeFields = Constants.Conventions.Member.GetStandardPropertyTypeStubs().Keys;
+
+        // include everything except the above
+        return new MemberValueSetValidator(null, null, null, excludeFields);
+    }
+}
+```
+
+Another example could be to replace the default Lucene analyzer for the ExternalIndex:
+
+```c#
+using System.Collections.Generic;
+using Examine;
+using Lucene.Net.Analysis;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Services;
+using Umbraco.Examine;
+using Umbraco.Web.Search;
+
+// override the default creator
+public class CustomUmbracoIndexesCreator : UmbracoIndexesCreator
+{
+    // ctor
+    public CustomUmbracoIndexesCreator(IProfilingLogger profilingLogger, ILocalizationService languageService, IPublicAccessService publicAccessService, IMemberService memberService, IUmbracoIndexConfig umbracoIndexConfig)
+        : base(profilingLogger, languageService, publicAccessService, memberService, umbracoIndexConfig)
+    {
+    }
+
+    // override the Create method
+    public override IEnumerable<IIndex> Create()
+    {
+        // create the default index definitions
+        var defaultIndexes = base.Create().ToDictionary(x => x.Name, x => x);
+        var internalIndex = defaultIndexes[Constants.UmbracoIndexes.InternalIndexName];
+        var memberIndex = defaultIndexes[Constants.UmbracoIndexes.MembersIndexName];
+
+        // create a custom external index
+        var externalIndex = CreateExternalIndex();
+
+        // return the Umbraco indexes - all 3 must be here with the correct names!
+        return new IIndex[]
+        {
+            internalIndex,
+            externalIndex,
+            memberIndex
+        };
+    }
+
+    private IIndex CreateExternalIndex()
+    {
+        var index = new UmbracoContentIndex(
+            Constants.UmbracoIndexes.ExternalIndexName, // default
+            CreateFileSystemLuceneDirectory(Constants.UmbracoIndexes.ExternalIndexPath), // default
+            new UmbracoFieldDefinitionCollection(), // default
+
+            // change the default analyzer from StandardAnalyzer
+            new WhitespaceAnalyzer(),
+
+            ProfilingLogger, // default
+            LanguageService, // default
+            UmbracoIndexConfig.GetPublishedContentValueSetValidator()); // default
+
+        return index;
+    }
+}
+```
+
+## Index based on document types
+
+The following example will show how to create an index that will only include nodes based on the document type _product_.
+
+:::note
+We always recommend that you use the existing built in ExternalIndex. You should then query based on the NodeTypeAlias instead of creating a new separate index based on that particular node type. However, should the need arise, the example below will show you how to do it.
```

Comment:
```suggestion
We always recommend that you use the existing built-in ExternalIndex. You should then query based on the NodeTypeAlias instead of creating a new separate index based on that particular node type. However, should the need arise, the example below will show you how to do it.
```